ocw_oer_export/create_csv.py

```diff
@@ -0,0 +1,163 @@
+
+from .client import fetch_data_from_api
+from .data_handler import load_data_from_json
+from .constants import API_URL
+from .utilities import cleanup_curly_brackets, html_to_text, markdown_to_text
+
+logging.basicConfig(level=logging.INFO)
+logger = logging.getLogger(__name__)
+
+
+def create_ocw_topic_to_oer_subject_mapping(path=None, file_name=None):
+    """Create a mapping between OCW topics and OER subjects."""
+    if path is None:
+        path = os.path.dirname(__file__)
+
+    if file_name is None:
+        file_name = "mapping_files/ocw_topic_to_oer_subject.csv"
+
+    file_path = os.path.join(path, file_name)
+    ocw_topic_to_subject = pd.read_csv(file_path)
+    ocw_topics_mapping = ocw_topic_to_subject.set_index("OCW Topic")[
+        "OER Subject"
+    ].to_dict()
+    return ocw_topics_mapping
+
+
+def get_cr_subjects(ocw_topics_mapping, ocw_course_topics):
+    """Get Course Resource Subjects based on OCW topics."""
+    oer_subjects_list = [
+        ocw_topics_mapping.get(topic["name"]).split("|")
+        if ocw_topics_mapping.get(topic["name"]) is not None
+        else []
+        for topic in ocw_course_topics
+    ]
+    unique_oer_subjects = set(
+        subject for subjects in oer_subjects_list for subject in subjects
+    )
+    return "|".join(unique_oer_subjects)
+
+
+def get_cr_keywords(list_of_topics_objs):
+    """Get Course Resource Keywords from a list of OCW topic objects."""
+    return "|".join(topic["name"] for topic in list_of_topics_objs)
+
+
+def get_cr_authors(list_of_authors_objs):
+    """Get Course Resource Authors from a list of OCW author objects."""
+    return "|".join(
+        f"{author['last_name']}, {author['first_name']}"
+        for author in list_of_authors_objs
+    )
+
+
+def get_cr_educational_use(course_feature_tags):
+    """Get Course Resource Educational Use based on course feature tags."""
+    tags = ["Curriculum/Instruction"]
+    assessment_flag = any("Assignment" in tag for tag in course_feature_tags)
+    professional_dev_flag = "Instructor Insights" in course_feature_tags
+
+    if assessment_flag:
+        tags.append("Assessment")
+
+    if professional_dev_flag:
+        tags.append("Professional Development")
+
+    return "|".join(tags)
+
+
+def get_cr_accessibility(course_feature_tags):
+    """Get Course Resource Accessibility tags based on course feature tags."""
+    tags = ["Visual|Textual"]
+    video_flag = any("Video" in tag for tag in course_feature_tags)
+
+    if video_flag:
+        tags.append("Auditory|Caption|Transcript")
+
+    return "|".join(tags)
+
+
+def get_description_in_plain_text(description):
+    """Get Course Resource plain text description by cleaning up markdown and HTML."""
+    stripped_markdown = markdown_to_text(description)
+    stripped_html = html_to_text(stripped_markdown)
+    plain_description = cleanup_curly_brackets(stripped_html)
+    return plain_description
+
+
+def process_single_course(course, ocw_topics_mapping):
```

Comment:
**Suggestion:** Rather than `process_single_course`, may I suggest `transform_single_course`? In the sense of [extract, transform, load](https://en.wikipedia.org/wiki/Extract,_transform,_load)? We tend to use that language elsewhere (e.g., in Open https://github.com/mitodl/mit-open/tree/main/learning_resources/etl)